src/utils.ts

```diff
@@ -410,23 +417,27 @@ export function formatProjectHash(multihash: Uint8Array): string {
 }
 
 // Identify an address by checking whether it's a contract or an externally-owned address.
-export async function identifyAddress(address: string, config: Config): Promise<AddressType> {
-  const safe = await isSafe(address, config);
-  if (safe) {
-    return AddressType.Safe;
-  }
+export const identifyAddress = memoizeLru(
+  async (address: string, config: Config): Promise<AddressType> => {
+    const safe = await isSafe(address, config);
+    if (safe) {
+      return AddressType.Safe;
+    }
 
-  const code = await config.provider.getCode(address);
-  const bytes = ethers.utils.arrayify(code);
+    const code = await config.provider.getCode(address);
+    const bytes = ethers.utils.arrayify(code);
 
-  if (bytes.length > 0) {
-    if (ethers.utils.keccak256(bytes) === config.orgs.contractHash) {
-      return AddressType.Org;
+    if (bytes.length > 0) {
+      if (ethers.utils.keccak256(bytes) === config.orgs.contractHash) {
+        return AddressType.Org;
+      }
+      return AddressType.Contract;
     }
-    return AddressType.Contract;
-  }
-  return AddressType.EOA;
-}
+    return AddressType.EOA;
+  },
+  (address: string) => address,
+  { max: 1000 }
```

Comment:
We should set a TTL here.

Reply:
I think thats not necessary, since the type of an address never changes, so having a 1000 entries as max value and that the least recent drops out, is good enough, also we only store `[address]: u8` key pairs

Also I feel we even should consider removing some TTLs since `lru-cache` does not remove them actively from the cache, so they still add to the size of the cache, but they are considered stale after the time passed.
> Max time to live for items before they are considered stale. Note that stale items are NOT preemptively removed by default, and MAY live in the cache, contributing to its LRU max, long after they have expired.

Reply:
> Also I feel we even should consider removing some TTLs since lru-cache does not remove them actively from the cache, so they still add to the size of the cache, but they are considered stale after the time passed.

I'd keep the other TTLs just for the sake of getting the user more up-to-date data.